ondevice/explainer.md

```diff
@@ -0,0 +1,76 @@
+
+- **Privacy concerns:** Raw and transcribed audio is transmitted over the network.
+- **Latency issues:** Users may experience delays due to network communication.
+- **Offline limitations:** Speech recognition does not work without an internet connection.
+
+To address these issues, we introduce **on-device speech recognition capabilities** as part of the Web Speech API. This enhancement allows speech recognition to run locally on user devices, providing a faster, more private, and offline-compatible experience.
+
+## Why Use On-Device Speech Recognition?
+
+### 1. **Privacy**
+On-device processing ensures that neither raw audio nor transcriptions leave the user's device, enhancing data security and user trust.
+
+### 2. **Performance**
+Local processing reduces latency, providing a smoother and faster user experience.
+
+### 3. **Offline Functionality**
+Applications can offer speech recognition capabilities even without an active internet connection, increasing their utility in remote or low-connectivity environments.
+
+## New Methods
+
+### 1. `Promise<boolean> onDeviceWebSpeechAvailable(DOMString lang)`
+This method checks if on-device speech recognition is available for a specific language. Developers can use this to determine whether to enable on-device features or fall back to cloud-based recognition.
+
+#### Example Usage
+```javascript
+const lang = 'en-US';
+SpeechRecognition.onDeviceWebSpeechAvailable(lang).then((available) => {
+    if (available) {
+        console.log(`On-device speech recognition is available for ${lang}.`);
+    } else {
+        console.log(`On-device speech recognition is not available for ${lang}.`);
+    }
+});
+```
+
+### 2. `Promise<boolean> installOnDeviceSpeechRecognition()`
+This method initiates the installation of resources required for on-device speech recognition. The installation process may download and configure necessary language models.
+
+#### Example Usage
+```javascript
+SpeechRecognition.installOnDeviceSpeechRecognition().then(() => {
+    console.log('Installation of on-device speech recognition resources initiated successfully.');
+}).catch((error) => {
+    console.error('Unable to install on-device speech recognition:', error);
+});
+```
+
+## New Attribute
+
+### 1. `mode` attribute in the `SpeechRecognition` interface
+The `mode` attribute in the `SpeechRecognition` interface defines how speech recognition should behave when starting a session.
+
+#### `SpeechRecognitionMode` Enum
+
+- **"ondevice-preferred"**: Use on-device speech recognition if available. If not, fall back to cloud-based speech recognition.
+- **"ondevice-only"**: Only use on-device speech recognition. If it's unavailable, throw an error.
+- **"cloud-only"**: Only use cloud-based speech recognition, bypassing on-device options entirely.
```

Comment:
I think these should be on-device-preferred and on-device-only. We don't have too many precedents on the web, but I can't see why we would remove the hyphens here.